homeassistant/components/template/update.py

```diff
@@ -0,0 +1,462 @@
+        self._title_template = config.get(CONF_TITLE)
+        self._update_percentage_template = config.get(CONF_UPDATE_PERCENTAGE)
+
+        self._attr_supported_features = UpdateEntityFeature(0)
+        if config[CONF_BACKUP]:
+            self._attr_supported_features |= UpdateEntityFeature.BACKUP
+        if config[CONF_SPECIFIC_VERSION]:
+            self._attr_supported_features |= UpdateEntityFeature.SPECIFIC_VERSION
+        if (
+            self._in_progress_template is not None
+            or self._update_percentage_template is not None
+        ):
+            self._attr_supported_features |= UpdateEntityFeature.PROGRESS
+
+        self._optimistic_in_process = (
+            self._in_progress_template is None
+            and self._update_percentage_template is not None
+        )
+
+    @callback
+    def _update_installed_version(self, result: Any) -> None:
+        if result is None:
+            self._attr_installed_version = None
+            return
+
+        self._attr_installed_version = cv.string(result)
+
+    @callback
+    def _update_latest_version(self, result: Any) -> None:
+        if result is None:
+            self._attr_latest_version = None
+            return
+
+        self._attr_latest_version = cv.string(result)
+
+    @callback
+    def _update_in_process(self, result: Any) -> None:
+        try:
+            self._attr_in_progress = cv.boolean(result)
+        except vol.Invalid:
+            _LOGGER.error(
+                "Received invalid in_process value: %s for entity %s.  Expected: True, False",
+                result,
+                self.entity_id,
+            )
+            self._attr_in_progress = False
+
+    @callback
+    def _update_release_summary(self, result: Any) -> None:
+        if result is None:
+            self._attr_release_summary = None
+            return
+
+        self._attr_release_summary = cv.string(result)
+
+    @callback
+    def _update_release_url(self, result: Any) -> None:
+        if result is None:
+            self._attr_release_url = None
+            return
+
+        try:
+            self._attr_release_url = cv.configuration_url(result)
+        except vol.Invalid:
+            _LOGGER.error(
+                "Received invalid release_url: %s for entity %s",
+                result,
+                self.entity_id,
+            )
+            self._attr_release_url = None
+
+    @callback
+    def _update_title(self, result: Any) -> None:
+        if result is None:
+            self._attr_title = None
+            return
+
+        self._attr_title = cv.string(result)
+
+    @callback
+    def _update_update_percentage(self, result: Any) -> None:
+        if result is None:
+            if self._optimistic_in_process:
+                self._attr_in_progress = False
+            self._attr_update_percentage = None
+            return
+
+        try:
+            percentage = vol.All(vol.Coerce(float), vol.Range(0, 100, True, True))(
```

Comment:
It's not clear what the bools do, please pass the min_included + max_included by name.